### DISCUSSION TITLE
Unable to create new project

### DISCUSSION THREAD

**Discussion by pkconcept** [symfony/symfony#52756] [Q&A] [answered]
### Symfony version(s) affected

6.3.*

### Description

Unable to create new project 
!!  PHP Fatal error:  Declaration of Symfony\Component\Cache\CacheItem::expiresAt(?DateTimeInterface $expiration): Symfony\Component\ErrorHandler\DebugClassLoader must be compatible with PsrExt\Cache\CacheItemInterface::expiresAt($expiration) in 

### How to reproduce

symfony new appDepot --version="6.3.*" --webapp                                                                                                 

### Possible Solution

fix bug ? :)

**Comment by xabbuh** [comment]
Your issue comes from using the [PHP PSR extension](https://github.com/jbboehr/php-psr). It looks like it only provides the interfaces for `psr/cache` 1.0 but the Cache component requires version 2 or 3 of `psr/cache`. There's nothing we can do in Symfony about it.

**Comment by derrabus** [comment]
Duplicate of #49984. Don't use the PSR extension.

**Comment by pkconcept** [comment]
Sorry but it's a symfony bug, using php 8.2.2 as standard, and can't create a project...
symphony is incompatible with 8.2.2

**Comment by xabbuh** [comment]
No, the PSR extension is not part of the PHP core. You need to uninstall it.

**Comment by pkconcept** [comment]
PSR extension is available as a standard extension , so everyone can install and use it.
Symfony install must check before proceed ...

**Comment by pkconcept** [comment]
My fresh php install , for a new project :

php8.2 - server-side, HTML-embedded scripting language (metapackage)
php8.2-bcmath - Bcmath module for PHP
php8.2-bz2 - bzip2 module for PHP
php8.2-cli - command-line interpreter for the PHP scripting language
php8.2-common - documentation, examples and common module for PHP
php8.2-curl - CURL module for PHP
php8.2-decimal - Arbitrary precision floating-point decimal for PHP
php8.2-dev - Files for PHP8.2 module development
php8.2-fpm - server-side, HTML-embedded scripting language (FPM-CGI binary)
php8.2-gd - GD module for PHP
php8.2-gmp - GMP module for PHP
php8.2-http - PECL HTTP module for PHP Extended HTTP Support
php8.2-igbinary - igbinary PHP serializer
php8.2-imagick - Provides a wrapper to the ImageMagick library
php8.2-imap - IMAP module for PHP
php8.2-inotify - Inotify bindings for PHP
php8.2-interbase - Interbase module for PHP
php8.2-intl - Internationalisation module for PHP
php8.2-ldap - LDAP module for PHP
php8.2-mailparse - Email message manipulation for PHP
php8.2-mbstring - MBSTRING module for PHP
php8.2-memcache - memcache extension module for PHP
php8.2-memcached - memcached extension module for PHP, uses libmemcached
php8.2-mongodb - MongoDB driver for PHP
php8.2-msgpack - PHP extension for interfacing with MessagePack
php8.2-mysql - MySQL module for PHP
php8.2-oauth - OAuth 1.0 consumer and provider extension
php8.2-odbc - ODBC module for PHP
php8.2-opcache - Zend OpCache module for PHP
php8.2-pcov - Code coverage driver
php8.2-pgsql - PostgreSQL module for PHP
php8.2-pspell - pspell module for PHP
php8.2-psr - PSR interfaces for PHP
php8.2-raphf - raphf module for PHP
php8.2-readline - readline module for PHP
php8.2-redis - PHP extension for interfacing with Redis
php8.2-smbclient - PHP wrapper for libsmbclient
php8.2-snmp - SNMP module for PHP
php8.2-soap - SOAP module for PHP
php8.2-sqlite3 - SQLite3 module for PHP
php8.2-ssh2 - Bindings for the libssh2 library
php8.2-stomp - Streaming Text Oriented Messaging Protocol (STOMP) client module for PHP
php8.2-sybase - Sybase module for PHP
php8.2-tidy - tidy module for PHP
php8.2-uploadprogress - file upload progress tracking extension for PHP
php8.2-uuid - PHP UUID extension
php8.2-xdebug - Xdebug Module for PHP
php8.2-xhprof - Hierarchical Profiler for PHP 5.x
php8.2-xml - DOM, SimpleXML, XML, and XSL module for PHP
php8.2-xmlrpc - XML-RPC servers and clients functions for PHP
php8.2-xsl - XSL module for PHP (dummy)
php8.2-yac - YAC (Yet Another Cache) for PHP
php8.2-yaml - YAML-1.1 parser and emitter for PHP
php8.2-zip - Zip module for PHP
php8.2-zmq - ZeroMQ messaging bindings for PHP
php8.2-zstd - Zstandard extension for PHP

**Comment by xabbuh** [comment]
This is simply not possible and there is nothing we can do about this rather than telling you to uninstall this extension.

**Comment by derrabus** [comment]
> PSR extension is available as a standard extension

It's an extension someone published on PECL. There's nothing "standard" about it.

This extension is defective. Please just don't use it.

**Comment by pkconcept** [comment]
Ok, removed ... but still not working .. sory..

11 package suggestions were added by new dependencies, use `composer suggest` to see details.
Generating autoload files
115 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
> symfony-cmd
sh: 1: symfony-cmd: not found
Script symfony-cmd handling the auto-scripts event returned with error code 127
Script @auto-scripts was called via post-update-cmd

                                                                         
  unable to run /usr/local/bin/composer require webapp --no-interaction

**Reply by ghost** [reply]
Install `symfony/flex` package to allow executing composer scripts of type `symfony-cmd`

**Reply by pkconcept** [reply]
It's not the good way ... Installing from scratch must work without artifacts ...
I've resolved by installing with debug mode, but it's not the best solution... Maybe symfony teams take a look ... one day ? ... to fix it ?

**Reply by ghost** [reply]
How did you get `scripts.auto-scripts.<command-name>` entry in you composer.json?
If it was not added by you then it must have been added automatically but with flex together. Or maybe someone removed/disabled Symfony Flex.
Find the step that adds this entry to your composer.json and let us know.

**Reply by pkconcept** [reply]
It's from a fresh install .

like this : symfony new my_project_directory --version="6.3.*" --webapp

need to run symfony new my_project_directory --version="6.3.*" --webapp --debug 
to make the install working ... composer.json is built from the symfony installer

**Reply by xabbuh** [reply]
Please post the complete output that you get when running `symfony new my_project_directory --version="6.3.*" --webapp`.

**Reply by ghost** [reply]
Working for me inside composer:2.6 container

![image](https://github.com/symfony/symfony/assets/25783196/660aba48-9acf-4740-b72e-3f707cb0eafb)

It must be something specific to you local environment.

**Reply by pkconcept** [reply]
#symfony new my_project_directory --version="6.3.*" --webapp


* Creating a new Symfony 6.3.* project with Composer
  (running /usr/local/bin/composer create-project symfony/skeleton /home/webdev/aclog/my_project_directory 6.3.* --no-interaction)

* Setting up the project under Git version control
  (running git init /home/webdev/aclog/my_project_directory)

  (running /usr/local/bin/composer require webapp --no-interaction)

Composer plugins have been disabled for safety in this non-interactive session. Set COMPOSER_ALLOW_SUPERUSER=1 if you want to allow plugins to run as root/super user.
Do not run Composer as root/super user! See https://getcomposer.org/root for details
./composer.json has been updated
Running composer update symfony/webapp-pack
Loading composer repositories with package information
Updating dependencies
Lock file operations: 102 installs, 0 updates, 0 removals
  - Locking doctrine/cache (2.2.0)
  - Locking doctrine/collections (2.1.4)
  - Locking doctrine/common (3.4.3)
  - Locking doctrine/dbal (3.7.2)
  - Locking doctrine/deprecations (1.1.2)
  - Locking doctrine/doctrine-bundle (2.11.1)
  - Locking doctrine/doctrine-migrations-bundle (3.3.0)
  - Locking doctrine/event-manager (2.0.0)
  - Locking doctrine/inflector (2.0.8)
  - Locking doctrine/instantiator (2.0.0)
  - Locking doctrine/lexer (2.1.0)
  - Locking doctrine/migrations (3.7.1)
  - Locking doctrine/orm (2.17.1)
  - Locking doctrine/persistence (3.2.0)
  - Locking doctrine/sql-formatter (1.1.3)
  - Locking egulias/email-validator (4.0.2)
  - Locking masterminds/html5 (2.8.1)
  - Locking monolog/monolog (3.5.0)
  - Locking myclabs/deep-copy (1.11.1)
  - Locking nikic/php-parser (v4.17.1)
  - Locking phar-io/manifest (2.0.3)
  - Locking phar-io/version (3.2.1)
  - Locking phpdocumentor/reflection-common (2.2.0)
  - Locking phpdocumentor/reflection-docblock (5.3.0)
  - Locking phpdocumentor/type-resolver (1.7.3)
  - Locking phpstan/phpdoc-parser (1.24.4)
  - Locking phpunit/php-code-coverage (9.2.29)
  - Locking phpunit/php-file-iterator (3.0.6)
  - Locking phpunit/php-invoker (3.1.1)
  - Locking phpunit/php-text-template (2.0.4)
  - Locking phpunit/php-timer (5.0.3)
  - Locking phpunit/phpunit (9.6.13)
  - Locking psr/clock (1.0.0)
  - Locking psr/link (2.0.1)
  - Locking sebastian/cli-parser (1.0.1)
  - Locking sebastian/code-unit (1.0.8)
  - Locking sebastian/code-unit-reverse-lookup (2.0.3)
  - Locking sebastian/comparator (4.0.8)
  - Locking sebastian/complexity (2.0.2)
  - Locking sebastian/diff (4.0.5)
  - Locking sebastian/environment (5.1.5)
  - Locking sebastian/exporter (4.0.5)
  - Locking sebastian/global-state (5.0.6)
  - Locking sebastian/lines-of-code (1.0.3)
  - Locking sebastian/object-enumerator (4.0.4)
  - Locking sebastian/object-reflector (2.0.4)
  - Locking sebastian/recursion-context (4.0.5)
  - Locking sebastian/resource-operations (3.0.3)
  - Locking sebastian/type (3.2.1)
  - Locking sebastian/version (3.0.2)
  - Locking symfony/asset (v6.3.8)
  - Locking symfony/browser-kit (v6.3.8)
  - Locking symfony/clock (v6.3.4)
  - Locking symfony/css-selector (v6.3.2)
  - Locking symfony/debug-bundle (v6.3.2)
  - Locking symfony/debug-pack (v1.0.10)
  - Locking symfony/doctrine-bridge (v6.3.8)
  - Locking symfony/doctrine-messenger (v6.3.8)
  - Locking symfony/dom-crawler (v6.3.4)
  - Locking symfony/expression-language (v6.3.0)
  - Locking symfony/form (v6.3.8)
  - Locking symfony/http-client (v6.3.8)
  - Locking symfony/http-client-contracts (v3.4.0)
  - Locking symfony/intl (v6.3.7)
  - Locking symfony/mailer (v6.3.5)
  - Locking symfony/maker-bundle (v1.51.1)
  - Locking symfony/messenger (v6.3.7)
  - Locking symfony/mime (v6.3.5)
  - Locking symfony/monolog-bridge (v6.3.8)
  - Locking symfony/monolog-bundle (v3.10.0)
  - Locking symfony/notifier (v6.3.6)
  - Locking symfony/options-resolver (v6.3.0)
  - Locking symfony/orm-pack (v2.4.0)
  - Locking symfony/password-hasher (v6.3.8)
  - Locking symfony/phpunit-bridge (v6.3.8)
  - Locking symfony/polyfill-intl-icu (v1.28.0)
  - Locking symfony/polyfill-intl-idn (v1.28.0)
  - Locking symfony/process (v6.3.4)
  - Locking symfony/profiler-pack (v1.0.6)
  - Locking symfony/property-access (v6.3.2)
  - Locking symfony/property-info (v6.3.0)
  - Locking symfony/security-bundle (v6.3.8)
  - Locking symfony/security-core (v6.3.7)
  - Locking symfony/security-csrf (v6.3.2)
  - Locking symfony/security-http (v6.3.8)
  - Locking symfony/serializer (v6.3.8)
  - Locking symfony/serializer-pack (v1.3.0)
  - Locking symfony/stopwatch (v6.3.0)
  - Locking symfony/test-pack (v1.1.0)
  - Locking symfony/translation (v6.3.7)
  - Locking symfony/translation-contracts (v3.4.0)
  - Locking symfony/twig-bridge (v6.3.8)
  - Locking symfony/twig-bundle (v6.3.8)
  - Locking symfony/twig-pack (v1.0.1)
  - Locking symfony/validator (v6.3.8)
  - Locking symfony/web-link (v6.3.0)
  - Locking symfony/web-profiler-bundle (v6.3.8)
  - Locking symfony/webapp-pack (v1.2.0)
  - Locking theseer/tokenizer (1.2.2)
  - Locking twig/extra-bundle (v3.8.0)
  - Locking twig/twig (v3.8.0)
  - Locking webmozart/assert (1.11.0)
Writing lock file
Installing dependencies from lock file (including require-dev)
Package operations: 102 installs, 0 updates, 0 removals
    0 [>---------------------------]    0 [->--------------------------]
  - Installing symfony/stopwatch (v6.3.0): Extracting archive
  - Installing doctrine/event-manager (2.0.0): Extracting archive
  - Installing doctrine/deprecations (1.1.2): Extracting archive
  - Installing doctrine/cache (2.2.0): Extracting archive
  - Installing doctrine/dbal (3.7.2): Extracting archive
  - Installing doctrine/migrations (3.7.1): Extracting archive
  - Installing doctrine/persistence (3.2.0): Extracting archive
  - Installing symfony/doctrine-bridge (v6.3.8): Extracting archive
  - Installing doctrine/sql-formatter (1.1.3): Extracting archive
  - Installing doctrine/doctrine-bundle (2.11.1): Extracting archive
  - Installing doctrine/doctrine-migrations-bundle (3.3.0): Extracting archive
  - Installing doctrine/lexer (2.1.0): Extracting archive
  - Installing doctrine/instantiator (2.0.0): Extracting archive
  - Installing doctrine/inflector (2.0.8): Extracting archive
  - Installing doctrine/common (3.4.3): Extracting archive
  - Installing doctrine/collections (2.1.4): Extracting archive
  - Installing doctrine/orm (2.17.1): Extracting archive
  - Installing symfony/polyfill-intl-idn (v1.28.0): Extracting archive
  - Installing egulias/email-validator (4.0.2): Extracting archive
  - Installing webmozart/assert (1.11.0): Extracting archive
  - Installing phpstan/phpdoc-parser (1.24.4): Extracting archive
  - Installing phpdocumentor/reflection-common (2.2.0): Extracting archive
  - Installing phpdocumentor/type-resolver (1.7.3): Extracting archive
  - Installing phpdocumentor/reflection-docblock (5.3.0): Extracting archive
  - Installing sebastian/version (3.0.2): Extracting archive
  - Installing sebastian/type (3.2.1): Extracting archive
  - Installing sebastian/resource-operations (3.0.3): Extracting archive
  - Installing sebastian/recursion-context (4.0.5): Extracting archive
  - Installing sebastian/object-reflector (2.0.4): Extracting archive
  - Installing sebastian/object-enumerator (4.0.4): Extracting archive
  - Installing sebastian/global-state (5.0.6): Extracting archive
  - Installing sebastian/exporter (4.0.5): Extracting archive
  - Installing sebastian/environment (5.1.5): Extracting archive
  - Installing sebastian/diff (4.0.5): Extracting archive
  - Installing sebastian/comparator (4.0.8): Extracting archive
  - Installing sebastian/code-unit (1.0.8): Extracting archive
  - Installing sebastian/cli-parser (1.0.1): Extracting archive
  - Installing phpunit/php-timer (5.0.3): Extracting archive
  - Installing phpunit/php-text-template (2.0.4): Extracting archive
  - Installing phpunit/php-invoker (3.1.1): Extracting archive
  - Installing phpunit/php-file-iterator (3.0.6): Extracting archive
  - Installing theseer/tokenizer (1.2.2): Extracting archive
  - Installing nikic/php-parser (v4.17.1): Extracting archive
  - Installing sebastian/lines-of-code (1.0.3): Extracting archive
  - Installing sebastian/complexity (2.0.2): Extracting archive
  - Installing sebastian/code-unit-reverse-lookup (2.0.3): Extracting archive
  - Installing phpunit/php-code-coverage (9.2.29): Extracting archive
  - Installing phar-io/version (3.2.1): Extracting archive
  - Installing phar-io/manifest (2.0.3): Extracting archive
  - Installing myclabs/deep-copy (1.11.1): Extracting archive
  - Installing phpunit/phpunit (9.6.13): Extracting archive
  - Installing psr/clock (1.0.0): Extracting archive
  - Installing psr/link (2.0.1): Extracting archive
  - Installing masterminds/html5 (2.8.1): Extracting archive
  - Installing symfony/dom-crawler (v6.3.4): Extracting archive
  - Installing symfony/browser-kit (v6.3.8): Extracting archive
  - Installing symfony/css-selector (v6.3.2): Extracting archive
  - Installing twig/twig (v3.8.0): Extracting archive
  - Installing symfony/translation-contracts (v3.4.0): Extracting archive
  - Installing symfony/twig-bridge (v6.3.8): Extracting archive
  - Installing symfony/debug-bundle (v6.3.2): Extracting archive
  - Installing symfony/http-client-contracts (v3.4.0): Extracting archive
  - Installing symfony/clock (v6.3.4): Extracting archive
  - Installing symfony/messenger (v6.3.7): Extracting archive
  - Installing monolog/monolog (3.5.0): Extracting archive
  - Installing symfony/monolog-bridge (v6.3.8): Extracting archive
  - Installing symfony/options-resolver (v6.3.0): Extracting archive
  - Installing symfony/password-hasher (v6.3.8): Extracting archive
  - Installing symfony/phpunit-bridge (v6.3.8): Extracting archive
  - Installing symfony/polyfill-intl-icu (v1.28.0): Extracting archive
  - Installing symfony/property-info (v6.3.0): Extracting archive
  - Installing symfony/security-core (v6.3.7): Extracting archive
  - Installing symfony/security-csrf (v6.3.2): Extracting archive
  - Installing symfony/property-access (v6.3.2): Extracting archive
  - Installing symfony/security-http (v6.3.8): Extracting archive
  - Installing symfony/serializer (v6.3.8): Extracting archive
  - Installing symfony/twig-bundle (v6.3.8): Extracting archive
  - Installing symfony/web-profiler-bundle (v6.3.8): Extracting archive
  - Installing symfony/web-link (v6.3.0): Extracting archive
  - Installing symfony/validator (v6.3.8): Extracting archive
  - Installing twig/extra-bundle (v3.8.0): Extracting archive
  - Installing symfony/twig-pack (v1.0.1): Extracting archive
  - Installing symfony/translation (v6.3.7): Extracting archive
  - Installing symfony/test-pack (v1.1.0): Extracting archive
  - Installing symfony/serializer-pack (v1.3.0): Extracting archive
  - Installing symfony/security-bundle (v6.3.8): Extracting archive
  - Installing symfony/profiler-pack (v1.0.6): Extracting archive
  - Installing symfony/process (v6.3.4): Extracting archive
  - Installing symfony/orm-pack (v2.4.0): Extracting archive
  - Installing symfony/notifier (v6.3.6): Extracting archive
  - Installing symfony/monolog-bundle (v3.10.0): Extracting archive
  - Installing symfony/mime (v6.3.5): Extracting archive
  - Installing symfony/maker-bundle (v1.51.1): Extracting archive
  - Installing symfony/mailer (v6.3.5): Extracting archive
  - Installing symfony/intl (v6.3.7): Extracting archive
  - Installing symfony/http-client (v6.3.8): Extracting archive
  - Installing symfony/form (v6.3.8): Extracting archive
  - Installing symfony/expression-language (v6.3.0): Extracting archive
  - Installing symfony/doctrine-messenger (v6.3.8): Extracting archive
  - Installing symfony/debug-pack (v1.0.10): Extracting archive
  - Installing symfony/asset (v6.3.8): Extracting archive
  - Installing symfony/webapp-pack (v1.2.0): Extracting archive
   0/102 [>---------------------------]   0%
  19/102 [=====>----------------------]  18%
  29/102 [=======>--------------------]  28%
  39/102 [==========>-----------------]  38%
  49/102 [=============>--------------]  48%
  52/102 [==============>-------------]  50%
  62/102 [=================>----------]  60%
  78/102 [=====================>------]  76%
  88/102 [========================>---]  86%
  92/102 [=========================>--]  90%
 102/102 [============================] 100%
11 package suggestions were added by new dependencies, use `composer suggest` to see details.
Generating autoload files
115 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
> symfony-cmd
sh: 1: symfony-cmd: not found
Script symfony-cmd handling the auto-scripts event returned with error code 127
Script @auto-scripts was called via post-update-cmd

                                                                         
  unable to run /usr/local/bin/composer require webapp --no-interaction

**Reply by pkconcept** [reply]
PHP 8.2 , debian Buster

**Reply by pkconcept** [reply, marked as answer]
hmm ok, found why... "root user" ... :)

**Comment by pkconcept** [comment]
only works if --debug passed ... why ? ...
